pkg/ai/types_ai.go

```diff
@@ -154,3 +118,266 @@ type AiNotificationsDestinationFilter struct {
+	// OAuth2 access token url
+	AccessTokenURL string `json:"accessTokenUrl"`
+	// Authentication Type - Token or Oauth2
+	AuthType AiNotificationsAuthType `json:"authType"`
+	// OAuth2 authorization url
+	AuthorizationURL string `json:"authorizationUrl"`
+	// OAuth2 clientId
+	ClientId string `json:"clientId"`
+	// Token prefix
+	Prefix string `json:"prefix"`
+	// Interval of how often should the access token be refreshed
+	RefreshInterval int `json:"refreshInterval,omitempty"`
+	// Is the OAuth2 access token refreshable
+	Refreshable bool `json:"refreshable"`
+	// OAuth2 token's scope
+	Scope string `json:"scope,omitempty"`
+}
+
+// GetAccessTokenURL returns a pointer to the value of AccessTokenURL from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetAccessTokenURL() string {
+	return x.AccessTokenURL
+}
+
+// GetAuthType returns a pointer to the value of AuthType from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetAuthType() AiNotificationsAuthType {
+	return x.AuthType
+}
+
+// GetAuthorizationURL returns a pointer to the value of AuthorizationURL from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetAuthorizationURL() string {
+	return x.AuthorizationURL
+}
+
+// GetClientId returns a pointer to the value of ClientId from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetClientId() string {
+	return x.ClientId
+}
+
+// GetPrefix returns a pointer to the value of Prefix from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetPrefix() string {
+	return x.Prefix
+}
+
+// GetRefreshInterval returns a pointer to the value of RefreshInterval from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetRefreshInterval() int {
+	return x.RefreshInterval
+}
+
+// GetRefreshable returns a pointer to the value of Refreshable from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetRefreshable() bool {
+	return x.Refreshable
+}
+
+// GetScope returns a pointer to the value of Scope from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetScope() string {
+	return x.Scope
+}
+
+// AiNotificationsAuth - Authentication interface
+type AiNotificationsAuth struct {
+	// OAuth2 access token url
+	AccessTokenURL string `json:"accessTokenUrl,omitempty"`
+	// Authentication Type - Token, Oauth2, or Basic
+	AuthType AiNotificationsAuthType `json:"authType,omitempty"`
+	// OAuth2 authorization url
+	AuthorizationURL string `json:"authorizationUrl,omitempty"`
+	// OAuth2 clientId
+	ClientId string `json:"clientId,omitempty"`
+	// Token prefix
+	Prefix string `json:"prefix,omitempty"`
+	// Interval of how often should the OAuth2 access token be refreshed
+	RefreshInterval int `json:"refreshInterval,omitempty"`
+	// Is the OAuth2 access token refreshable
+	Refreshable bool `json:"refreshable,omitempty"`
+	// OAuth2 token's scope
+	Scope string `json:"scope,omitempty"`
+	// Basic auth password
+	Password SecureValue `json:"password,omitempty"`
```

Comment:
I think we can remove the password here - in the API we don't return it back:
<img width="280" alt="Screen Shot 2022-07-12 at 17 06 32" src="https://user-images.githubusercontent.com/69894591/178509471-7a065723-1143-49d1-91fa-dcfc13a412b0.png">

Reply:
Good catch, you're right. It's only needed for the mutation inputs 🙂